sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Emitter.cs

```diff
@@ -1,98 +1,74 @@
-                    AddRegistrationExtension(indentedTextWriter);
-
-                    indentedTextWriter.Indent--;
-                    indentedTextWriter.Write("}");
-
-                    indentedTextWriter.Flush();
-                }
-
-                return stringWriter.ToString();
-            }
-
-            private void SetUpUsings(IndentedTextWriter indentedTextWriter, bool hasHttpTrigger)
-            {
-                indentedTextWriter.WriteLine("// <auto-generated/>");
-                indentedTextWriter.WriteLine("using System;");
-                indentedTextWriter.WriteLine("using System.Collections.Generic;");
-                indentedTextWriter.WriteLine("using System.Collections.Immutable;");
-                indentedTextWriter.WriteLine("using System.Text.Json;");
-                indentedTextWriter.WriteLine("using System.Threading.Tasks;");
-                indentedTextWriter.WriteLine("using Microsoft.Azure.Functions.Worker.Core.FunctionMetadata;");
-                indentedTextWriter.WriteLine("using Microsoft.Extensions.DependencyInjection;");
-                indentedTextWriter.WriteLine("using Microsoft.Extensions.Hosting;");
-
-                if (hasHttpTrigger)
-                {
-                    indentedTextWriter.WriteLine("using Microsoft.Azure.Functions.Worker.Http;");
-                }
-            }
-
-            private void WriteGetFunctionsMetadataAsyncMethod(IndentedTextWriter indentedTextWriter, IReadOnlyList<GeneratorFunctionMetadata> functionMetadata, CancellationToken cancellationToken)
-            {
-                indentedTextWriter.WriteLine("public Task<ImmutableArray<IFunctionMetadata>> GetFunctionMetadataAsync(string directory)");
-                indentedTextWriter.WriteLine("{");
-                indentedTextWriter.Indent++;
-
-                // create list of IFunctionMetadata and populate it
-                indentedTextWriter.WriteLine("var metadataList = new List<IFunctionMetadata>();");
-                AddFunctionMetadataInfo(indentedTextWriter, functionMetadata, cancellationToken);
-                indentedTextWriter.WriteLine("return Task.FromResult(metadataList.ToImmutableArray());");
-
-                indentedTextWriter.Indent--;
-                indentedTextWriter.WriteLine("}");
+                string functionMetadataInfo = AddFunctionMetadataInfo(funcMetadata, cancellationToken);
+
+                return $$"""
+                         // <auto-generated/>
+                         using System;
+                         using System.Collections.Generic;
+                         using System.Collections.Immutable;
+                         using System.Text.Json;
+                         using System.Threading.Tasks;
+                         using Microsoft.Azure.Functions.Worker.Core.FunctionMetadata;
+                         using Microsoft.Azure.Functions.Worker.Http;
```

Comment:
Did you end up having to keep this import?

Reply:
I need to make sure we have good validation. Will keep it as part of these changes and send up update for the removal.